tests/common/from_params_test.py

```diff
@@ -862,3 +862,93 @@ def __init__(self, nested_class: Optional[Union[str, NestedClass]] = None):
             WrapperClass.from_params(
                 params=Params({"nested_class": {"wrong_varname": "varstring"}})
             )
+
+    def test_from_params_handles_base_class_kwargs(self):
+        class Foo(FromParams):
+            def __init__(self, a: int, b: str = None, **kwargs) -> None:
+                self.a = a
+                self.b = b
+                for key, value in kwargs.items():
+                    setattr(self, key, value)
+
+        foo = Foo.from_params(Params({"a": 2, "b": "hi"}))
+        assert foo.a == 2
+        assert foo.b == "hi"
+
+        foo = Foo.from_params(Params({"a": 2, "b": "hi", "c": {"2": "3"}}))
+        assert foo.a == 2
+        assert foo.b == "hi"
+        assert foo.c == {"2": "3"}
```

Comment:
I think we're missing a test case where we have a `FromParams`  base class like `Foo` here that takes `**kwargs`, and then a subclass of `Foo` that takes `**kwargs` and just passes them on to its super class constructor:

```suggestion
        assert foo.c == {"2": "3"}
        
        class Bar(Foo):
            def __init__(self, a: int,  b: str, d: int, **kwargs) -> None:
                super().__init__(a, b=b, **kwargs)
                self.d = d
                
        bar = Bar.from_params(Params({"a": 2, "b": "hi", "c": {"2": "3"}, "d": 0}))
        assert bar.a == 2
        assert bar.b == "hi"
        assert bar.c == {"2": "3"}
        assert bar.d == 0
```

Reply:
Hmm, yeah, I didn't think to check that.  Looks like it already passes, though.